File: sdkwork-ai-sdk-flutter/lib/src/api/embeddings.dart
import '../http/client.dart';
import '../models.dart';

class EmbeddingsApi {
  final HttpClient _client;
  
  EmbeddingsApi(this._client);

  /// Create embeddings
  Future<EmbeddingResponse?> createEmbedding(EmbeddingRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/embeddings'), body: body, contentType: 'application/json');
    return response is EmbeddingResponse ? response : null;
  }
}

File: sdkwork-ai-sdk-python/sdkwork_ai_sdk/api/embeddings.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import EmbeddingRequest, EmbeddingResponse

class EmbeddingsApi:
    """Embeddings API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def create_embedding(self, body: EmbeddingRequest) -> EmbeddingResponse:
        """Create embeddings"""
        return self._client.post(f"/ai/v3/v1/embeddings", json=body)
